src/main/java/edu/cornell/kfs/concur/service/impl/ConcurAccountValidationServiceImpl.java

```diff
@@ -2,17 +2,173 @@
 
 import java.util.ArrayList;
 
+import org.kuali.kfs.coa.businessobject.Account;
+import org.kuali.kfs.coa.businessobject.ObjectCode;
+import org.kuali.kfs.coa.businessobject.ProjectCode;
+import org.kuali.kfs.coa.businessobject.SubAccount;
+import org.kuali.kfs.coa.businessobject.SubObjectCode;
+import org.kuali.kfs.coa.service.AccountService;
+import org.kuali.kfs.coa.service.ObjectCodeService;
+import org.kuali.kfs.coa.service.ProjectCodeService;
+import org.kuali.kfs.coa.service.SubAccountService;
+import org.kuali.kfs.coa.service.SubObjectCodeService;
+import org.kuali.kfs.sys.KFSKeyConstants;
+import org.kuali.rice.core.api.mo.common.active.Inactivatable;
+
+import edu.cornell.kfs.concur.ConcurConstants;
 import edu.cornell.kfs.concur.businessobjects.ConcurAccountInfo;
 import edu.cornell.kfs.concur.businessobjects.ValidationResult;
 import edu.cornell.kfs.concur.service.ConcurAccountValidationService;
+import edu.cornell.kfs.sys.service.ErrorMessageUtilsService;
 
 public class ConcurAccountValidationServiceImpl implements ConcurAccountValidationService {
+    protected AccountService accountService;
+    protected ObjectCodeService objectCodeService;
+    protected SubAccountService subAccountService;
+    protected SubObjectCodeService subObjectCodeService;
+    protected ProjectCodeService projectCodeService;
+    protected ErrorMessageUtilsService errorMessageUtilsService; 
 
     @Override
     public ValidationResult validateConcurAccountInfo(ConcurAccountInfo concurAccountInfo) {
-        ValidationResult validationResult = new ValidationResult(false, new ArrayList<String>());
+        return checkAccountingString(
+                concurAccountInfo.getChart(),
+                concurAccountInfo.getAccountNumber(),
+                concurAccountInfo.getSubAccountNumber(),
+                concurAccountInfo.getObjectCode(),
+                concurAccountInfo.getSubObjectCode(),
+                concurAccountInfo.getProjectCode());
+    }
+
+    public ValidationResult checkAccountingString(String chartOfAccountsCode, String accountNumber, String subAccountNumber, String objectCode, String subObjectCode, String projectCode) {
+        ValidationResult requiredFieldsValidationResult = checkRequiredAccountInfo(chartOfAccountsCode, accountNumber, objectCode);
+
+        if (requiredFieldsValidationResult.isNotValid()) {
+            return requiredFieldsValidationResult;
+        } else {
+            ValidationResult accountValidationResult = checkAccount(chartOfAccountsCode, accountNumber);
+            if (accountValidationResult.isNotValid()) {
+                return accountValidationResult;
+            } else {           
+                ValidationResult validationResult = new ValidationResult(true, new ArrayList<String>());
```

Comment:
I would say this section could still be extracted into another method and then there would just be the one line call in this else. If there aren't other changes to make, I probably wouldn't insist on just changing this though.